### DISCUSSION TITLE
Unwanted parts are generated

### DISCUSSION THREAD

**Discussion by yanger216** [lllyasviel/Fooocus#3461] [Q&A] [answered]
hi everyone, i'm occurred a problem when i was trying to change the background of a cosmetic bottle. everything was good. But somethings were generated and added to bottle. Here are the input image and some output. Also i tried to add some negative prompts such as 'do not expand the original bottle', but it didn't work.
![image](https://github.com/user-attachments/assets/4b6ebadc-eb8b-43ef-8eeb-3f4c0f03f406)
![image](https://github.com/user-attachments/assets/b537fb31-c151-4956-9421-56a4428a6d61)
So could you guys tell me what should i do to avoid it happen again, thank you for your advice.

**Comment by mashb1t** [comment, marked as answer]
Please add the same image to image prompt and use PyraCanny to optimize following the shape of the input + enable mixing of inpaint and image prompt in developer debug mode.
You may also erode the mask by 1px and test again.

**Reply by yanger216** [reply]
thanks for your answer. it's quite useful. But i found that the background and the bottle in the generated image look very **abrupt**. Can i do something to make it look smoother?

**Reply by mashb1t** [reply]
no, there's currently no way to make a smooth transition between mask and background.

**Reply by yanger216** [reply]
Thanks for your patient reply. Appreciate it!
